framework/src/org/checkerframework/common/fieldtypeinference/FieldTypeInferenceAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,42 @@
+package org.checkerframework.common.fieldtypeinference;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.framework.flow.CFAbstractAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+
+/**
+ * The ATF of a Checker must be a subtype of this class for it to perform
+ * intra-class inference for private fields.
+ *
+ * @author pbsf
+ *
+ */
+public class FieldTypeInferenceAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
+
+    /** Maps a Receiver representing a private field to an ATM that contains the
+    LUB of all AMs they have been assigned to. */
+    private final Map<Receiver, AnnotatedTypeMirror> assignedPrivateFields =
```

Comment:
We want to refine private fields only while typechecking a source file right?  That is, if I have MyClass.java, I want to keep track of all assignments for every field in MyClass.java.  But once I have typechecked MyClass.java, I don't need this information anymore?  Correct?  If so, then shouldn't we clear this map once the compilation unit has changed (see AnnotatedTypeFactory.setRoot).  Otherwise, we will have a map of ALL private fields in the program which may be a problem on large programs and auto-generated code.